### PR TITLE
Clean up and improve Viterbi decoder

### PR DESCRIPTION
**Pull request by jacobmcnamee**

### CODE REVIEW

**Comment by ljbade** [comment]
@jacobmcnamee The code looks fine to me as far as formatting and such. I trust that you are confident the code now does what it is supposed to do.

You should merge.

File: tests/check_viterbi.c [hunk 1]
```diff
@@ -9,83 +9,91 @@
...
+  v27_decision_t decisions[HISTORY_LENGTH_BITS];
+  v27_poly_t v27_poly;
+  signed char poly_bytes[] = {V27POLYA, V27POLYB};
+  int v27_bit_length = -32; /* hack to process 32 bits before outputting data */
+
+  v27_poly_init(&v27_poly, poly_bytes);
+  v27_init(&v, decisions, HISTORY_LENGTH_BITS, &v27_poly, 0);
+
+  /* Perform decoding, writing output to file */
+  int output = open(tmp_file, O_WRONLY | O_CREAT, 0644);
```
Comment (gsmcmullin) [inline tests/check_viterbi.c]: Why do we write the output to a file and then compare files, rather than just using memory buffers?

**Comment by gsmcmullin** [comment]
This looks like a good improvement.  Go ahead and merge after resolving conflicts.

**Pull request merged**